aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,10 +1,171 @@
+
+Bundling and minification primarily improve the first page request load time. Once a web page has been requested, the browser caches the assets (JavaScript, CSS and images) so bundling and minification won’t provide any performance boost when requesting the same page, or pages on the same site requesting the same assets. If you don’t set the expires header correctly on your assets, and you don’t use bundling and minification, the browsers freshness heuristics will mark the assets stale after a few days and the browser will require a validation request for each asset. In this case, bundling and minification provide a performance increase even after the first page request. 
+
+Bundling
+--------
+
+Bundling is a feature that makes it easy to combine or bundle multiple files into a single file. Because bundling combines multiple files into a single file, it reduces the number of requests to the server that is required to retrieve and display a web asset, such as a web page. You can create CSS, JavaScript and other bundles. Fewer files, means fewer HTTP requests from your browser to the server or from the service providing your application. This results in improved first page load performance.
+
+Bundling can be accomplished using the `gulp-concat <https://www.npmjs.com/package/gulp-concat>`__ task, which is installed via the Node Package Manager (`npm <https://www.npmjs.com/>`__). Add the ``gulp-concat`` package to the ``devDependencies`` section of your ``packages.json`` file:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/package.json
+  :language: json
+  :linenos:
+  :emphasize-lines: 6
+
+Run ``npm install`` to then install the specified packages.
+
+.. note:: Visual Studio hides the package.json file by default. To edit package.json right-click on the npm node under Dependencies in the solution explorer and select **Open package.json**. Visual Studio will automatically install npm packages whenever package.json is modified.
+
+In your ``gulpfile.js`` import the ``gulp-concat`` module:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/gulpfile.js
+  :language: js
+  :linenos:
+  :lines: 4-8
+  :emphasize-lines: 3
+
+Use globbing patterns to specify the files that you want to bundle and minify:
+
+.. literalinclude:: /../common/samples/WebApplication1/src/WebApplication1/gulpfile.js
+  :language: js
+  :linenos:
+  :lines: 14-19
+
+You can then define gulp tasks that run ``concat`` on the desired files and output the result to your webroot:
```

Comment:
Can we point to something that explains the code?  You chain (pipe) the commands. What's the ! all about? I can guess paths..css and path.minCss